linker/shared/src/test/scala/org/scalajs/linker/testutils/TestIRRepo.scala

```diff
@@ -13,22 +13,54 @@
 package org.scalajs.linker.testutils
 
 import scala.concurrent._
-import scala.concurrent.ExecutionContext.Implicits.global
 
 import org.scalajs.linker.StandardImpl
 import org.scalajs.linker.interface.IRFile
 
 object TestIRRepo {
   private val globalIRCache = StandardImpl.irFileCache()
 
-  val minilib: Future[Seq[IRFile]] = load(StdlibHolder.minilib)
-  val javalib: Future[Seq[IRFile]] = load(StdlibHolder.javalib)
+  val minilib: Future[Seq[IRFile]] = loadGlobal(StdlibHolder.minilib)
+  val javalib: Future[Seq[IRFile]] = loadGlobal(StdlibHolder.javalib)
   val empty: Future[Seq[IRFile]] = Future.successful(Nil)
-  val previousLibs: Map[String, Future[Seq[IRFile]]] =
-    StdlibHolder.previousLibs.map(x => x._1 -> load(x._2))
 
-  private def load(stdlibPath: String) = {
+  private def loadGlobal(stdlibPath: String): Future[Seq[IRFile]] = {
+    import scala.concurrent.ExecutionContext.Implicits.global
+
     Platform.loadJar(stdlibPath)
       .flatMap(globalIRCache.newCache.cached _)
   }
+
+  /** For each previous lib, calls `f(version, irFiles)`, and combines the result.
+   *
+   *  This method applies `f` *sequentially*. It waits until the returned
+   *  `Future` completes before moving on to the next iteration.
+   */
+  def sequentiallyForEachPreviousLib[A](f: (String, Seq[IRFile]) => Future[A])(
+      implicit ec: ExecutionContext): Future[List[A]] = {
+
+    def loop(previousLibs: List[(String, String)]): Future[List[A]] = {
```

Comment:
Have you considered putting this into a helper that does just the sequential processing? IMO this is harder to read than it needs to be.

Reply:
Ah, indeed, that is definitely worth extracting.